src/components/chain-selector/index.tsx

```diff
@@ -0,0 +1,287 @@
+                fontFamily: " 'Lato' ",
+                fontWeight: 700,
+                fontSize: "18px",
+                lineHeight: '26px',
+                marginBottom: '10px',
+                marginTop: '43px'
+            }
+        },
+        addBtn: {
+            fontWeight: 800,
+            fontSize: '18px',
+            lineHeight: '22px',
+            padding: '10px 40px',
+            marginLeft: '20px'
+        }
+    }),
+);
+
+const ChainSelector: React.FC = observer(() => {
+    const { chainInfo, setChainInfo } = useStores().chainInfoStore
+
+    const implementChain = async (chain: Chain) => {
+        const currency: Currency = {
+            ...chainInfo.stakeCurrency,
+            coinDenom: chain.coinDenom,
+            coinMinimalDenom: chain.coinMinimalDenom
+        }
+        const updatedChainInfo: ChainInfo = {
+            ...chainInfo,
+            ...chain,
+            stakeCurrency: currency,
+            currencies: [{ ...currency }],
+            feeCurrencies: [{ ...currency }]
+        }
+
+        await setChainInfo(updatedChainInfo);
+    }
+
+    const handleChainChanged = (event: React.ChangeEvent<{ value: unknown }>) => {
+        const chainId = event.target.value as string
+        if (chainId === "+") return // skip "+ new chain" option
+
+        const chain = chainListProvider().find((chain: Chain) => chain.chainId === chainId)
+        implementChain(chain)
+    }
+
+    const handleAddChain = () => {
+        // todo: do we need to save in localstore all added chains?
+        // todo check for the uniq name. We are using name as key for items, it should be uniq or we need to change a key
+        implementChain(newChain)
+        setAllChainList( [...chainListProvider(), newChain])
+        setOpen(false)
+    }
+
+    const [allChainList, setAllChainList] = React.useState([...chainListProvider(), chainInfo]);
+    const [open, setOpen] = React.useState(false);
+    const [newChain, setNewChain] = React.useState<Chain | undefined>(undefined);
+    const [presetChainId, setPresetChainId] = React.useState<string | undefined>(undefined);
+
+    const handlePresetChanged = (event: React.ChangeEvent<{ value: unknown }>) => {
+        const chainId = event.target.value as string
+        const presetChainInfo = chainListProvider().find((chain: Chain) => chain.chainId === chainId)
+
+        setPresetChainId(presetChainInfo.chainId)
+        setNewChain({...presetChainInfo})
+    }
+
+    const handleOpen = async () => {
+        setNewChain({
+                chainId: chainInfo.chainId,
+                chainName: chainInfo.chainName,
+                rpc: chainInfo.rpc,
+                rest: chainInfo.rest,
+                coinDenom: chainInfo.stakeCurrency.coinDenom,
+                coinMinimalDenom: chainInfo.stakeCurrency.coinMinimalDenom
+        })
+        setPresetChainId(chainInfo.chainId)
+        setOpen(true);
+    };
+
+    const handleClose = () => {
+        setOpen(false);
```

Comment:
we don't use semicolons
And I think it's a good time to configure linter for this